### PR TITLE
fix: use virtual import for storage drivers

### PR DESCRIPTION
**Pull request by ascorbic**
## Changes

Currently, storage drivers are imported directly from `unstorage` subpaths. This causes a problem in sites that use pnpm, because its just a transitive dependency. This PR changes it so that at build time it imports a virtual module instead, which resolves to the full path of the module, ensuring that it's bundled.

## Testing

Tested with Node and Netlify adapters

### CODE REVIEW

**Comment by ascorbic** [comment]
!preview

**Comment by ascorbic** [comment]
!preview sessions

**Comment by ascorbic** [comment]
!preview sessions

File: packages/astro/src/core/session.ts [hunk 1]
```diff
@@ -363,21 +363,30 @@ export class AstroSession<TDriver extends SessionDriverName = any> {
...
+		const isBuiltin = this.#config.driver in builtinDrivers;
+		// Try to load the driver from the built-in unstorage drivers.
+		// Otherwise, assume it's a custom driver and load by name.
+		const driverPackage: string = isBuiltin
+			? builtinDrivers[this.#config.driver as keyof typeof builtinDrivers]
+			: this.#config.driver;
 
-			driver = await import(entry).then((r) => r.default || r);
+		try {
+			// If driver is not a builtin, and in development and test, load the driver directly.
```
Comment (ematipico) [inline packages/astro/src/core/session.ts:374]:
```suggestion
			// If driver is not a builtin, or in development or test, load the driver directly.
```
It seems the comment doesn't match the condition below

File: packages/astro/src/vite-plugin-sessions/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			}
+		},
+		async load(id) {
+			if (id === RESOLVED_SESSION_DRIVER_ID) {
+				let driver = settings.config.experimental?.session?.driver;
+				if (driver && driver in builtinDrivers) {
+					if (driver === 'fs') {
+						// fs tries to bundle chokidar (and therefore fsevents), which is a binary
+						driver = 'fsLite';
+					}
```
Comment (ematipico) [inline packages/astro/src/vite-plugin-sessions/index.ts:29]: I wonder if the users should be aware of this, maybe with a warning or something?

File: packages/astro/src/core/session.ts (hunk 1 above)
Comment (ascorbic) [reply packages/astro/src/core/session.ts:374]: Yeah it's poorly worded. It meant "a and b and c are situations where this applies"

**Pull request merged**